test_conformance/api/test_queue_array_properties.cpp

```diff
@@ -0,0 +1,273 @@
+                            NULL);
+    test_error(
+        error,
+        "clGetDeviceInfo for CL_DEVICE_DEVICE_ENQUEUE_CAPABILITIES failed");
+    if (!(device_enqueue_caps & CL_DEVICE_QUEUE_SUPPORTED)
+        && requested_size > 0)
+    {
+        log_info("\tCL_DEVICE_QUEUE_SUPPORTED false, skipped\n");
+        return TEST_SKIPPED_ITSELF;
+    }
+    for (auto check_property : all_properties)
+    {
+        if (check_property & requested_bitfield)
+        {
+            if (check_property & on_host_properties)
+            {
+                log_info("\t%s supported\n",
+                         GetQueuePropertyName(check_property));
+            }
+            else if (device_enqueue_caps & CL_DEVICE_QUEUE_SUPPORTED)
+            {
+                if ((check_property & CL_QUEUE_ON_DEVICE)
+                    || (check_property & CL_QUEUE_ON_DEVICE_DEFAULT))
+                {
+                    log_info("\t%s supported\n",
+                             GetQueuePropertyName(check_property));
+                }
+            }
+            else
+            {
+                log_info("\t%s not supported, skipped\n",
+                         GetQueuePropertyName(check_property));
+                return TEST_SKIPPED_ITSELF;
+            }
+        }
+    }
+    return error;
+}
+
+int create_queue_and_check_array_properties(
+    cl_context context, cl_device_id deviceID,
+    clCommandQueueWrapper& test_queue,
+    test_queue_array_properties_data test_case)
+{
+    int error = CL_SUCCESS;
+    if (test_case.properties.size() > 0)
+    {
+        test_queue = clCreateCommandQueueWithProperties(
+            context, deviceID, test_case.properties.data(), &error);
+        test_error(error, "clCreateCommandQueueWithProperties failed");
+    }
+    else
+    {
+        test_queue =
+            clCreateCommandQueueWithProperties(context, deviceID, NULL, &error);
+        test_error(error, "clCreateCommandQueueWithProperties failed");
+    }
+    std::vector<cl_queue_properties> get_properties;
+    size_t set_size = 0;
+
+    error = clGetCommandQueueInfo(test_queue, CL_QUEUE_PROPERTIES_ARRAY, 0,
+                                  NULL, &set_size);
+    test_error(
+        error,
+        "clGetCommandQueueInfo failed asking for CL_QUEUE_PROPERTIES_ARRAY");
+
+    if (test_case.properties.size() == 0 && set_size == 0)
+    {
+        return TEST_PASS;
+    }
+
+    cl_uint number_of_props = set_size / sizeof(cl_queue_properties);
+    get_properties.resize(number_of_props);
+    error = clGetCommandQueueInfo(test_queue, CL_QUEUE_PROPERTIES_ARRAY,
+                                  set_size, get_properties.data(), NULL);
+    test_error(
+        error,
+        "clGetCommandQueueInfo failed asking for CL_QUEUE_PROPERTIES_ARRAY");
+
+    if (get_properties.size() == 1 && get_properties[0] == 0
+        && test_case.properties.size() == 0)
+    {
+        return TEST_PASS;
+    }
```

Comment:
This case can be removed now, since we're requiring that if the incoming properties were `NULL` (for this test, indicated by the size of the properties vector being zero) that the query size return zero.

See: https://github.com/KhronosGroup/OpenCL-Docs/pull/398#issuecomment-672211863, #880.